### PR TITLE
Csv upload

### PR DESCRIPTION
**Pull request by sainingo**
# Work on API for uploading CSV files & The UI

Issue I worked on

see https://ampath.atlassian.net/browse/DAP-2

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/csvUploads/CsvUpload.tsx [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+      const response = await res.json();
+        console.log(response);
+
+      // use toast to display message
+      if (response.status === "success") {
+        //set timeout to reload page
+        toast.success(response.message);
+        setTimeout(() => {
+          window.location.reload();
+        }, 4000);
```
Comment (kipchubett) [inline src/components/csvUploads/CsvUpload.tsx]: Why the timeout?

File: src/components/Home/Home.tsx
Comment (kipchubett) [inline src/components/Home/Home.tsx:1]: This is a stale code, we no longer use this file for the home menu.

File: src/components/csvUploads/DisplayCSV.tsx [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+    }).then(async (willDelete) => {
+      if (willDelete) {
+        try {
+          const res = await deleteCsvFile(id);
+          const response = await res.json();
+          if (response.affectedRows === 1) {
+            toast.success("File deleted successfully");
+            setTimeout(() => {
+              window.location.reload();
+            }, 6000);
```
Comment (kipchubett) [inline src/components/csvUploads/DisplayCSV.tsx]: Why the timeout? 6s??
`async` - `await` should be sufficient in this case.

File: src/components/csvUploads/csv.resource.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+export interface UpdatedData {
+    status: string,
+    successful: number,
+    eid_file_upload_metadata_id: number,
+}
+
+
+const url = 'http://localhost:3000/api'
```
Comment (kipchubett) [inline src/components/csvUploads/csv.resource.ts]: The Base URL shouldn't be hardcoded

File: yarn.lock
Comment (kipchubett) [inline yarn.lock:1]: Can you recheck this file? It seems to me you are re-formatting every line

File: vite.config.ts [hunk 4]
```diff
@@ -5,22 +5,17 @@ import react from "@vitejs/plugin-react";
...
     proxy: {
-      "/ws": {
-        target: "https://ngx.ampath.or.ke/amrs",
-        changeOrigin: true,
-        secure: false,
-      },
-      "/api": {
-        target: "http://0.0.0.0:5080",
+      "/openmrs": {
+        target: "http://localhost:8089/",
```
Comment (kipchubett) [inline vite.config.ts]: For this project `openmrs` is part of the base URL
```
{
"/ws": {
        target: "http://localhost:8089/openmrs",
        changeOrigin: true,
        secure: false,
      },
}
```

**Review by kipchubett — CHANGES_REQUESTED** [review]
Thanks, @sainingo for your work on this
I've requested a few changes/comments.
But everything else looks good to me!

File: src/components/csvUploads/CsvUpload.tsx (hunk 1 above)
Comment (sainingo) [reply src/components/csvUploads/CsvUpload.tsx]: the reason for the timeout was to give certain duration b4 reloading the page but its not really neccesary

File: src/components/Home/Home.tsx
Comment (sainingo) [reply src/components/Home/Home.tsx:1]: :+1:

File: src/components/csvUploads/csv.resource.ts (hunk 3 above)
Comment (sainingo) [reply src/components/csvUploads/csv.resource.ts]: :+1:

File: src/components/csvUploads/DisplayCSV.tsx (hunk 2 above)
Comment (sainingo) [reply src/components/csvUploads/DisplayCSV.tsx]: :+1:

File: vite.config.ts (hunk 4 above)
Comment (sainingo) [reply vite.config.ts]: :+1:

**Review by kipchubett — APPROVED** [review]

**Pull request closed without merging**